api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -1236,13 +1236,33 @@ message UpstreamConnectionOptions {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.api.v2.UpstreamConnectionOptions";
 
+  enum FirstAddressFamilyVersion {
+    DEFAULT = 0;
+    V4 = 1;
+    V6 = 2;
+  }
+
+  message HappyEyeballsConfig {
+    // Specify the ip version of the first address being sorted in happy
+    // eyeballs.
+    FirstAddressFamilyVersion first_address_family_version = 1;
+
+    // Specify the count of addresses being interleaved by the sorting algorithm.
+    int32 first_address_family_count = 2;
```

Comment:
I think you can add `[(validate.rules).uint32 = {gt: 0}];` here to ensure that this is at least 1.